ingenialink/configuration_file.py

```diff
@@ -0,0 +1,458 @@
+import os
+import re
+from abc import ABC
+from typing import Optional, Union
+from xml.dom import minidom
+from xml.etree import ElementTree
+
+import ingenialogger
+import numpy as np
+
+from ingenialink import RegAccess, RegDtype
+from ingenialink.dictionary import Dictionary, Interface
+from ingenialink.exceptions import ILConfigurationFileParseError
+from ingenialink.register import Register
+
+logger = ingenialogger.get_logger(__name__)
+
+
+class Device:
+    """Device data for ConfigurationFile (XCF) class."""
+
+    interface: Interface
+    part_number: Optional[str]
+    product_code: Optional[int]
+    revision_number: Optional[int]
+    firmware_version: Optional[str]
+    node_id: Optional[int] = None
+
+    ELEMENT_NAME = "Device"
```

Comment:
All this is part of the implementation. They could be private

Reply:
I didn't make them private because they are used in the class method. In addition ELEMENT_NAME is used in ConfigurationFile too, to avoid write the string twice. I can make them protected. What do you think?

Reply:
Using private attributes in classmethods is ok. I think it's the whole purpose of classemthods, instead of using static methods or plain methods.

This does work:
`````
class C:


    __ATTR = "a"

    @classmethod
    def cm(cls):
        return cls.__ATTR
    

print(C.cm())
````